lib/external_service_new/zendesk_downloader.rb

```diff
@@ -0,0 +1,51 @@
+module ExternalServiceNew
+	class ZendeskDownloader
+		def initialize(dispatcher:, logger:nil)
+			@dispatcher = dispatcher
+			@logger			= logger || respond_to?(:logger) ? logger : nil
```

Comment:
Not sure about this logic,  ``` @logger  = logger ``` is sufficient. 
#respond_to? will be reached if logger is falsy (falsy == false || nil) and so #respond_to? will always return false